example/src/button-js/index.js

```diff
@@ -0,0 +1,44 @@
+import Component from '../../../src/index';
+
+class MyButton extends Component {
+  render() {
+    this.html `
+    <style>
+      :host {
+        --base-font-size: 24px;
+      }
+      :host button {
+        font-size: var(--base-font-size);
+        background: transparent;
+        border-radius: 3px;
+        border: 2px solid blue;
+        color: blue;
+        margin: 0 1em;
+        padding: 0.25em 1em;
+        outline: none;
+        position: relative;
+      }
+      :host button:focus {
+        border-style: dotted;
+      }
+      :host button:active {
+        top: 1px;
+        left: 1px;
+      }
+      :host button:hover {
+        border-style: solid;
+      }
+    </style>
+    <button onclick="${this.onButtonClick.bind(this)}">Normal Button</button>
```

Comment:
Binding inside render could be a memory issue. I would setup this binding inside the class constructor

```
constructor() {
        super();
        this.onButtonClick = this.onButtonClick.bind(this);
}
```